### DISCUSSION TITLE
过期：默认播放列表排序问题

### DISCUSSION THREAD

**Discussion by zfekin** [hooke007/mpv_PlayKit#210] [回收]
我有个文件夹里的文件命名是xxx-1至50这样的，默认的播放列表播放时排序是xxx-1,xxx-10到19，xxx-2,xxx-20到29....，请问怎么修改成从小到大的排序呢

**Comment by hooke007** [comment]
<s>这个问题上游已经修了，我在等定版后同步</s>

edit: 已同步

**Reply by zfekin** [reply]
我看到dyphire给出解决方法了，可是我在mpv-lazy文件夹里没找到他说的相关文件

**Reply by hooke007** [reply]
我已经告诉你原因了，uosc已经修了 
我还没同步

**Reply by dyphire** [reply]
我的错，没有注意到你是lite版用户，那个是由uosc控制的文件加载排序功能

**Reply by zfekin** [reply]
哦，好的，我想着我试试给修改了

**Comment by dyphire** [comment]
即使是autolad.lua(load_plus.lua)原本的排序算法也可以正常处理此案例，它不能处理的是系列字母数字中缺失数字时的特定排序问题

而这种完全错误的数字排序看上去像是lua原生字符串排序的结果，就本仓库而言我猜测可能是以下原因导致的：
https://github.com/hooke007/MPV_lazy/blob/19b0666a687fb5df10acc8049c4bb68921eed4c5/portable_config/script-opts/playlistmanager.conf#L29-L37
你可能在使用playlistmanager.lua的加载文件功能（`loadfiles_on_start=yes`）,但是却未启用`sortplaylist_on_start=yes`和`sortplaylist_on_file_add=yes`选项。导致脚本未使用自然排序算法而是使用了lua原生字符串排序

**Reply by hooke007** [reply]
现在的仓库里的东西不全是懒人包里的东西
（或者说大部分都不是
